packages/jazz-react/src/auth/DemoAuth.tsx

```diff
@@ -69,16 +69,29 @@ export function useDemoAuth({
 export const DemoAuthBasicUI = ({
   appName,
   state,
+  user,
 }: {
   appName: string;
   state: DemoAuthState;
+  user?: string;
 }) => {
   const [username, setUsername] = useState<string>("");
   const darkMode =
     typeof window !== "undefined"
       ? window.matchMedia("(prefers-color-scheme: dark)").matches
       : false;
 
+  if (user && state.state === "ready") {
+    if (state.existingUsers.includes(user)) {
+      state.logInAs(user);
+      return <></>;
+    } else {
+      state.signUp(user);
```

Comment:
[Side effects](https://react.dev/learn/keeping-components-pure#side-effects-unintended-consequences) should not run [in render](https://react.dev/reference/rules/components-and-hooks-must-be-pure#how-does-react-run-your-code), as React can render components multiple times to create the best possible user experience.

We could fix the issue using a `useEffect`:
```ts
const isAutoLogin = Boolean(user && state.state === "ready");

useEffect(() => {
  if (!isAutoLogin) return;
  
  if (state.existingUsers.includes(user)) {
      state.logInAs(user);
    } else {
      state.signUp(user);
    }
}, [isAutoLogin]);

if (isAutoLogin) return <></>;
```